lib/joke.rb

```diff
@@ -1,2 +1,16 @@
 class Joke
+  attr_reader :id, :question, :answer
+  def initialize(id, question, answer)
+    @id = id
+    @question = question
+    @answer = answer
+  end
+
+  def setup
+    @question
+  end
+
+  def punchline
+    @answer
+  end
```

Comment:
These last two methods are okay. It would have been easier to just call the instance variables `@setup` and `@punchline` instead of question/answer, and expose those with the attr_reader instead.